glib/src/variant_type.rs

```diff
@@ -35,6 +35,21 @@ impl VariantType {
     pub fn new(type_string: &str) -> Result<VariantType, BoolError> {
         VariantTy::new(type_string).map(ToOwned::to_owned)
     }
+
+    /// Tries to create a `VariantType` from an owned string.
+    ///
+    /// Returns `Ok` if the string is a valid type string, `Err` otherwise.
+    pub fn from_string(type_string: crate::GString) -> Result<VariantType, BoolError> {
```

Comment:
We could make it take an `Into<GString>`, but the reason is that it takes ownership of the string so no new allocation and no copying is necessary.

Reply:
FWIW some outlook to the future: for the release after this one, all functions that take `transfer full` string parameters would actually take an `impl Into<GString>` (and `transfer none` would be `impl Into<Cow<'a, GStr>>` where `GStr` is basically the same as `std::ffi::CStr` with UTF-8 guarantees, so that also in this case you wouldn't need to do any copying if you already have a `NUL`-terminated string).

Reply:
Let's go with `Into` then so this doesn't have to be changed later :)